File: interpreter/src/main/java/org/enso/interpreter/nodes/expression/operator/AddOperatorNode.java
package org.enso.interpreter.nodes.expression.operator;

import com.oracle.truffle.api.nodes.NodeInfo;
import org.enso.interpreter.nodes.expression.BinaryOperatorNode;

@NodeInfo(shortName = "+")
public abstract class AddOperatorNode extends BinaryOperatorNode {

    protected long add(long left, long right) {
        return Math.addExact(left, right);
    }
}

File: interpreter/src/stage0/scala/org/enso/interpreter/nodes/expression/operator/AddOperatorNode.scala
package org.enso.interpreter.nodes.expression.operator

import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.nodes.NodeInfo
import org.enso.interpreter.nodes.expression.BinaryOperatorNode

@NodeInfo(shortName = "+") abstract class AddOperatorNode
    extends BinaryOperatorNode {

  @Specialization protected def add(left: Long, right: Long): Long =
    Math.addExact(left, right)
}
